src/cli/commands/list.js

```diff
@@ -180,21 +181,33 @@ export function filterTree(tree: Tree, filters: Array<string>): boolean {
   return notDim && (found || hasChildren);
 }
 
+const getDevDeps = (manifest): Array<string> => Object.keys(manifest.devDependencies).map(key => `${key}@${manifest.devDependencies[key]}`);
+
 export async function run(config: Config, reporter: Reporter, flags: Object, args: Array<string>): Promise<void> {
   const lockfile = await Lockfile.fromDirectory(config.lockfileFolder, reporter);
   const install = new Install(flags, config, reporter, lockfile);
-  const {requests: depRequests, patterns, workspaceLayout} = await install.fetchRequestFromCwd();
+
+
+  const {requests: depRequests, patterns, manifest, workspaceLayout} = await install.fetchRequestFromCwd();
   await install.resolver.init(depRequests, {
     isFlat: install.flags.flat,
     isFrozen: install.flags.frozenLockfile,
     workspaceLayout,
   });
 
+  let pkgPatterns = [];
+  if (isProduction(process.env)) {
+    const devDeps = getDevDeps(manifest);
+    pkgPatterns = patterns.filter(pattern => !devDeps.includes(pattern));
```

Comment:
I would convert `devDeps` to a `Set` for faster look up. Technically, both can be sets since this is a set subtraction but the `Set` class doesn't have that method unless you use a third-party library.